File: backend/src/main/java/com/carrental/backend/vehicle/Vehicle.java
package com.carrental.backend.vehicle;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "vehicles")
@NoArgsConstructor
@Getter
@Setter
public class Vehicle {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String brand;
    private String model;
    private int year;

    private String type;
    private String driveType;

    private double pricePerDay;
    private boolean available;
    private String imageUrl;

    @Column(length = 1000)
    private String description;
}

File: frontend/src/types/Vehicle.ts
export interface Vehicle {
  id: number;
  brand: string;
  model: string;
  year: number;
  type: string;
  driveType: string;
  pricePerDay: number;
  available: boolean;
  imageUrl?: string;
  description?: string;
}

export interface VehicleFilter {
  type?: string;
  brand?: string;
  model?: string;
  driveType?: string;
  minPrice?: number;
  maxPrice?: number;
  sortBy?: string;
  sortDirection?: 'asc' | 'desc';
}
